lib/vrl/compiler/src/expression/query.rs

```diff
@@ -1,24 +1,25 @@
 use crate::expression::{Container, FunctionCall, Resolved, Variable};
 use crate::parser::ast::Ident;
-use crate::{Context, Expression, Path, State, TypeDef, Value};
+use crate::{Context, Expression, State, TypeDef, Value};
+use lookup::LookupBuf;
 use std::collections::BTreeMap;
 use std::fmt;
 
 #[derive(Clone, PartialEq)]
 pub struct Query {
     target: Target,
-    path: Path,
+    path: LookupBuf,
 }
 
 impl Query {
     // TODO:
     // - error when trying to index into object
     // - error when trying to path into array
-    pub fn new(target: Target, path: Path) -> Self {
+    pub fn new(target: Target, path: LookupBuf) -> Self {
         Query { target, path }
     }
 
-    pub fn path(&self) -> &Path {
+    pub fn path(&self) -> &LookupBuf {
```

Comment:
Isn't the idea that we'd return `Lookup` instead of a reference to a `LookupBuf`?

Reply:
I'm not sure I understand what the benefit of that would be? There would be additional allocations and processing to convert to a Lookup - everything apart from the actual field names which would be shared.